File: mobile/meowmentum/composeApp/src/commonMain/kotlin/org/meowmentum/project/domain/model/Task.kt
package org.meowmentum.project.domain.model

import kotlinx.datetime.Instant
import org.meowmentum.project.data.models.TaskPriority
import org.meowmentum.project.data.models.TaskStatus

data class Task(
    val id: Long = 0,
    val title: String,
    val description: String? = null,
    val createdAt: Instant? = null,
    val deadline: Instant? = null,
    val completedAt: Instant? = null,
    val status: TaskStatus = TaskStatus.PENDING,
    val priority: TaskPriority? = null,
    val tags: List<Tag>? = null,
    val timeIntervals: List<TimeInterval>? = null
)

File: backend/dotnet/Meowmentum.Server.Dotnet.Api/Meowmentum.Server.Dotnet.Core/Entities/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Meowmentum.Server.Dotnet.Core.Entities;

public class Task
{
    [Key]
    public long Id { get; set; }
    [Required]
    [MaxLength(200)]
    public required string Title { get; set; }
    public string? Description { get; set; }
    [Required]
    public DateTime CreatedAt { get; set; }
    public DateTime? Deadline { get; set; }
    public TaskStatus? Status { get; set; }
    public TaskPriority? Priority { get; set; } 
    [Required]
    [ForeignKey("User")]
    public long UserId { get; set; }
    public virtual ICollection<TimeInterval>? TimeIntervals { get; set; }
    public virtual ICollection<TaskTag>? TaskTags { get; set; }
}

public enum TaskStatus
{
    Pending,
    InProgress,
    Completed
}

public enum TaskPriority
{
    High,
    Medium,
    Low
}
